File: src/org/javaturk/oopj/ch01/MonteCarloPI.py
#
#  MonteCarloPI.py
#
#  Created by Akin Kaldiroglu on 2017.
#  Copyright 2017 JavaTurk. All rights reserved.

import random as r
import math as m
import time

# Total number of point to throw.
s = input("Please enter number of points: ")
n = int(s)

dotsInCircle = 0

start = time.clock()
# Iterate for the number of darts.
for i in range(0, n):
  # Generate random x, y in [0, 1].
  x = r.random()
  y = r.random()
  # Increment if inside unit circle.
  if (x*x + y*y) < 1.0:
      dotsInCircle += 1

end = time.clock()

# inside / total = pi / 4
pi = (float(dotsInCircle) / n) * 4

print "PI:            " , m.pi
print "Calculated PI: " , pi
seconds_elapsed = (end - start)
print "Time is ", seconds_elapsed, " s."

File: src/org/javaturk/oopj/ch01/MonteCarloPI.cpp
//
//  MonteCarloPI.cpp
//
//  Created by Akin Kaldiroglu on 18/09/15.
//  Copyright © 2015 JavaTurk. All rights reserved.

#include <iostream>
#include <stdlib.h> // srand
#include <ctime>

using namespace std;

# define PI_L  3.141592653589793238462643383279502884L

int main() {
    cout.precision(10);
    int n;
    int dotsInCircle = 0;
    cout << "Please enter number of points: " << endl;
    cin >> n;
    clock_t start = clock();
    for (int i = 0; i < n; i ++) {
        double x = static_cast<double>(rand()) / static_cast<double>(RAND_MAX);
        double y = static_cast<double>(rand()) / static_cast<double>(RAND_MAX);
        double distance = x * x + y * y;
        if (distance <= 1.0)
            dotsInCircle ++;
    }
    clock_t end = clock();
    float time =  ((float) end - start) / CLOCKS_PER_SEC;
    double myPi = (double) 4 * dotsInCircle / n;

    cout << "PI:            " << PI_L << endl;
    cout << "Calculated PI: " << myPi << endl;
    cout << "Time is: " << time << " s." <<endl;

    return 0;
}
